### DISCUSSION TITLE
Content templates for element types

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#11742] [Features and ideas]
Based on Umbraco 9.1.2 it seems it from settings is possible to create content templates for element types. Does this make sense? If it does and the UI support use of content templates for element types, there should be an configurable option to disable or enable this.

![image](https://user-images.githubusercontent.com/2919859/145112992-2f042fb6-9cb6-42d1-aede-dfa777e4fa3e.png)

In this specific project we don't need to create content templates for block settings element types.

**Comment by leekelleher** [comment]
On a side note, _(just in case anyone goes ahead to remove this for element types 😊)._ In my [Contentment's Content Blocks editor](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/content-blocks.md#how-to-configure-the-editor) I do make use of this feature, providing UI to create and use Content Templates for Element Types (at block level).

**Comment by bjarnef** [comment]
While investigating https://github.com/umbraco/Umbraco-CMS/discussions/11743 I found a way to remove Content Templates tree from Settings section and register it in Content section instead.

However currently there's a limitation to editors as they don't have access to document types tree, so they get an error in create dialog. Fixed in https://github.com/umbraco/Umbraco-CMS/pull/11796

Furthermore is may make sense to allow Content Templates to element types, but should it in some way be configurable?  E.g. we can potentially have a lot of element types used just for block settings.

Furthermore we can have composition document types (either used on regular document types or eleement types), e.g. SEO, Navigation and base  element type used on several blocks (a simple example in may contain background color and text color properties).
Should these be excluded? Maybe.

Currently out-of-the-box Umbraco only has the UI to create a Content Template from an existing page (document type) - at least for editors. Admins can create Content Templates for compositions and element types as well.

If https://github.com/umbraco/Umbraco-CMS/pull/11796 is merged, this will also be possible to editors.
Maybe it should as a start filter the content types further to non-editors or make it configurable. Ideally a configuration in appsettings.json or a property on a document type to configure it per content type. By default it need to be allowed to create a content template as it is the currrent behavior - then we could disable these ones which we don't need to create content templates for?

It would be much simpler if the editors only get 10 options instead of 50 due a lot of element types used in block content and setttings.

// cc @nielslyngsoe @madsrasmussen
